src/lib/snap_to.js

```diff
@@ -0,0 +1,120 @@
+const StringSet = require('../lib/string_set');
+const coordEach = require('@turf/meta').coordEach;
+const point = require('turf-point');
+const pointOnLine  = require('@turf/point-on-line');
+const distance = require('@turf/distance');
```

Comment:
working on this at https://github.com/andrewharvey/mapbox-gl-draw/tree/snap-to-edge-and-vertex